### DISCUSSION TITLE
[Question] Where does `auto` service get cuda?

### DISCUSSION THREAD

**Discussion by bonlime** [AbdBarho/stable-diffusion-webui-docker#233] [Q&A] [answered]
Hi, i'm trying to build some custom docker image with automatic inside and reading through your Dockerfiles i can't understand where CUDA is installed? I see that for `hlky & lstein` there is a line in Dockerfile with 
`RUN conda install cudatoolkit=11.6 -c conda-forge && conda clean -a -y`
but there is no mention of CUDA in automatic docker and still it somehow works. Could you point me where it gets the CUDA?

**Comment by AbdBarho** [comment, marked as answer]
@bonlime here:
https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/8623c737411e79fdeb80be713a75b2cdb87736f3/services/AUTOMATIC1111/Dockerfile#L37

which basically tells pip: _please install pytorch that comes bundled with cuda 11.6_

**Reply by bonlime** [reply]
Thanks, now I get it. I though it meant "download pytorch prebuilt for cuda 11.6" but didn't include the cuda itself.
